WooCommerce/Classes/ViewRelated/MainTabBarController.swift

```diff
@@ -603,9 +607,13 @@ private extension MainTabBarController {
             guard let self = self else { return }
             Task { @MainActor in
                 await self.showProductDetails(for: error)
+                self.analytics.track(event: .ImageUpload
+                    .failureSavingProductAfterImageUploadNoticeTapped(productOrVariation: error.productOrVariationEventProperty))
```

Comment:
Should we capture `self` weakly or with `unowned` here for extra safety?

Reply:
Makes sense, updated in https://github.com/woocommerce/woocommerce-ios/pull/7196/commits/820f10a8ec78aa15bb1050a47b073fd0e58bd2d6